### PR TITLE
[PLT-7793] Added /users/tokens endpoint

### PR DESCRIPTION
**Pull request by watadarkstar**
#### Summary
Adds a new endpoint called `/users/tokens` which gets all tokens for all users if one has the `manage_system` permission.

#### PR Links

- [Documentation PR](https://github.com/mattermost/mattermost-api-reference/pull/318)
- [Webapp PR](https://github.com/mattermost/mattermost-webapp/pull/496)
- [Server PR](https://github.com/mattermost/mattermost-server/pull/8038)
- Redux PR

#### Ticket Link

- [Jira](https://mattermost.atlassian.net/browse/PLT-7793)
- [Github](https://github.com/mattermost/mattermost-server/issues/7584#issuecomment-352808489)

#### Checklist
- [x] Added route
- [x] Added handler
- [x] Updated store
- [x] Added or updated unit tests (required for all new features)
- [x] [Added documentation](https://github.com/mattermost/mattermost-api-reference/pull/318)
- [x] Has UI changes
- [x] Touches critical sections of the codebase (access tokens)
- [x] Run `make test` for sanity check
- [x] Run `make check-style`
- [x] Tested with postman / curl
- [x] Rename end point to `/users/tokens`
- [x] Add i18n string

### CODE REVIEW

**Comment by watadarkstar** [comment]
@jwilander @saturninoabril I'm not really sure why my build is failing. Seems like its failing in `make test-te` but I'm not sure; from the [console output](https://build.mattermost.com/job/mattermost-platform-pr-split/1041/console) in Jenkins.

**Comment by ccbrown** [comment]
@watadarkstar We have a race condition that causes the builds to fail sometimes. I'm restarting it.

**Comment by watadarkstar** [comment]
Thanks @ccbrown I tried rebuilding it once but I guess it happened again.

File: model/client4.go [hunk 1]
```diff
@@ -1035,10 +1039,23 @@ func (c *Client4) CreateUserAccessToken(userId, description string) (*UserAccess
 	}
 }
 
-// GetUserAccessToken will get a user access token's id, description and the user_id
-// of the user it is for. The actual token will not be returned. Must have the
-// 'read_user_access_token' permission and if getting for another user, must have the
-// 'edit_other_users' permission.
+// GetAllUserAccessToken will get a all access token's id, description, is_active
```
Comment (jwilander) [inline model/client4.go]: Minor grammar and text updates `a all access token's` -> `a page of access tokens'`

File: model/client4.go [hunk 2]
```diff
@@ -1035,10 +1039,23 @@ func (c *Client4) CreateUserAccessToken(userId, description string) (*UserAccess
...
+// and the user_id in the system. The actual token will not be returned. Must have
+// the 'manage_system' permission.
+func (c *Client4) GetAllUserAccessTokens(page int, perPage int) ([]*UserAccessToken, *Response) {
```
Comment (jwilander) [inline model/client4.go]: Might want to remove the 'All' bit from here if it's removed from the endpoint

File: store/sqlstore/user_access_token_store.go [hunk 3]
```diff
@@ -171,6 +171,18 @@ func (s SqlUserAccessTokenStore) Get(tokenId string) store.StoreChannel {
 	})
 }
 
+func (s SqlUserAccessTokenStore) GetAll(offset, limit int) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		tokens := []*model.UserAccessToken{}
+
+		if _, err := s.GetReplica().Select(&tokens, "SELECT * FROM UserAccessTokens LIMIT :Limit OFFSET :Offset", map[string]interface{}{"Offset": offset, "Limit": limit}); err != nil {
+			result.Err = model.NewAppError("SqlUserAccessTokenStore.GetAll", "store.sql_user_access_token.get_all.app_error", nil, err.Error(), http.StatusInternalServerError)
```
Comment (jwilander) [inline store/sqlstore/user_access_token_store.go]: You'll need to add this localization string to the `i18n/en.json` file (not the other languages, those will be handled by the translators)

File: api4/user.go [hunk 4]
```diff
@@ -59,6 +59,7 @@ func (api *API) InitUser() {
 
 	api.BaseRoutes.User.Handle("/tokens", api.ApiSessionRequired(createUserAccessToken)).Methods("POST")
 	api.BaseRoutes.User.Handle("/tokens", api.ApiSessionRequired(getUserAccessTokens)).Methods("GET")
+	api.BaseRoutes.Users.Handle("/tokens/all", api.ApiSessionRequired(getAllUserAccessTokens)).Methods("GET")
```
[the remark this answers is not in the document]
Comment (watadarkstar) [reply api4/user.go]: Fixed

File: model/client4.go (hunk 1 above)
Comment (watadarkstar) [reply model/client4.go]: Fixed and also fixed in `GetUserAccessToken` below.

File: model/client4.go (hunk 2 above)
Comment (watadarkstar) [reply model/client4.go]: I renamed this to `GetUserAccessTokens` and have changed the previous definition of `getUserAccessTokens` to `getUserAccessTokensForUser` so I can use `getUserAccessTokens` and so the codebase is consistent.

File: store/sqlstore/user_access_token_store.go (hunk 3 above)
Comment (watadarkstar) [reply store/sqlstore/user_access_token_store.go]: Fixed

**Comment by watadarkstar** [comment]
@jwilander Fixed all your requested changes!

**Comment by jwilander** [comment]
Good stuff, we'll merge once the release branching is complete for 4.6

**Pull request merged**